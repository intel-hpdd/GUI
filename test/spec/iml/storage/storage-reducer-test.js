// @flow

import storageReducer, {
  ADD_STORAGE_RESOURCE_CLASSES,
  SET_STORAGE_SELECT_INDEX,
  ADD_STORAGE_RESOURCES,
  SET_STORAGE_CONFIG,
  SET_STORAGE_SORTING,
  SET_STORAGE_TABLE_LOADING
} from "../../../../source/iml/storage/storage-reducer.js";

describe("storageReducer", () => {
  it("should be a function", () => {
    expect(storageReducer).toEqual(expect.any(Function));
  });

  it("should return an initial state", () => {
    expect(storageReducer(undefined, { type: "FOO", payload: 3 })).toMatchSnapshot();
  });

  it("should add storage resource classes", () => {
    expect(
      storageReducer(undefined, {
        type: ADD_STORAGE_RESOURCE_CLASSES,
        payload: ["foo"]
      })
    ).toMatchSnapshot();
  });

  it("should storage resources", () => {
    expect(
      storageReducer(undefined, {
        type: ADD_STORAGE_RESOURCES,
        payload: { meta: {}, objects: [] }
      })
    ).toMatchSnapshot();
  });

  it("should set the storage config", () => {
    expect(
      storageReducer(undefined, {
        type: SET_STORAGE_CONFIG,
        payload: { entries: 20, offset: 100 }
      })
    ).toMatchSnapshot();
  });

  it("should set sorting", () => {
    expect(
      storageReducer(undefined, {
        type: SET_STORAGE_SORTING,
        payload: "foo"
      })
    ).toMatchSnapshot();
  });

  it("should set sorting desc", () => {
    const state1 = storageReducer(undefined, {
      type: SET_STORAGE_SORTING,
      payload: "foo"
    });

    const state2 = storageReducer(state1, {
      type: SET_STORAGE_SORTING,
      payload: "foo"
    });

    expect(state2).toMatchSnapshot();
  });

  it("should set the select index", () => {
    expect(
      storageReducer(undefined, {
        type: SET_STORAGE_SELECT_INDEX,
        payload: 3
      })
    ).toMatchSnapshot();
  });

  it("should set the loading", () => {
    expect(
      storageReducer(undefined, {
        type: SET_STORAGE_TABLE_LOADING,
        payload: true
      })
    ).toMatchSnapshot();
  });

  it("should return the same object on no loading change", () => {
    const state1 = storageReducer(undefined, {
      type: SET_STORAGE_TABLE_LOADING,
      payload: true
    });

    const state2 = storageReducer(state1, {
      type: SET_STORAGE_TABLE_LOADING,
      payload: true
    });

    expect(state1).toBe(state2);
  });
});
