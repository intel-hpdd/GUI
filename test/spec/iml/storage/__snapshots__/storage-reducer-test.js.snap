// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storageReducer should add storage resource classes 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": Array [
    "foo",
  ],
  "resources": null,
}
`;

exports[`storageReducer should return an initial state 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should set sorting 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "foo",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should set sorting desc 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": true,
    "sortKey": "foo",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should set the loading 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": true,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should set the select index 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": 3,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should set the storage config 1`] = `
Object {
  "config": Object {
    "entries": 20,
    "loading": false,
    "offset": 100,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": null,
  "resources": null,
}
`;

exports[`storageReducer should storage resources 1`] = `
Object {
  "config": Object {
    "entries": 10,
    "loading": false,
    "offset": 0,
    "selectIndex": null,
    "sortDesc": false,
    "sortKey": "",
  },
  "resourceClasses": null,
  "resources": Object {
    "meta": Object {},
    "objects": Array [],
  },
}
`;
